opendevin/runtime/client/README.md

```diff
@@ -7,32 +7,64 @@ This README provides an overview of the OpenDevin Runtime, a crucial component o
 
 The following sections dive deeper into these topics, providing a comprehensive understanding of the OpenDevin Runtime system.
 
+## Architecture Diagram
+
+```
++-------------------+     +-------------------+
+|   OpenDevin       |     |   Docker Host     |
+|   Backend         |     |                   |
+|                   |     |  +-------------+  |
+|  +-------------+  |     |  |  Runtime    |  |
+|  | EventStream |  |     |  |  Container  |  |
+|  | Runtime     |<-|-----|->|             |  |
+|  +-------------+  |     |  |  +-------+  |  |
+|                   |     |  |  |Runtime|  |  |
+|                   |     |  |  |Client |  |  |
+|                   |     |  |  +-------+  |  |
+|                   |     |  |     |       |  |
+|                   |     |  |  +-------+  |  |
+|                   |     |  |  |Plugins|  |  |
+|                   |     |  |  +-------+  |  |
+|                   |     |  +-------------+  |
++-------------------+     +-------------------+
+```
+
+This diagram illustrates the high-level architecture of the OpenDevin Runtime system:
+
+1. The OpenDevin Backend communicates with the Docker Host through the EventStreamRuntime.
+2. The Docker Host runs a Runtime Container, which includes:
+   - The Runtime Client: Handles incoming actions and generates observations.
+   - Plugins: Extend the functionality of the Runtime Client.
+3. The Runtime Client executes actions within the sandboxed environment of the Docker container.
+
+This architecture ensures a secure and flexible environment for executing AI-driven development tasks, allowing OpenDevin to execute a wide range of actions safely and efficiently.
+
 ## How the Runtime Image is Built
 
 The OpenDevin runtime uses a layered approach for building Docker images:
 
 1. **Original Image**: `ubuntu:22.04`
    - This is the base image used for all subsequent layers.
 
-2. **Runtime Image**:  `od_runtime:ubuntu__22.04`
+2. **Runtime Image**:  `od_runtime:od_v{OPENDEVIN_VERSION}_image_ubuntu__22.04`
```

Comment:
Maybe add one example for illustration with an actual version, too?

Reply:
Good point! Updated!